### ISSUE TITLE
unsafe-eval required when setting up CSP header

### ISSUE THREAD

**Issue by cmonnom** [molstar/molstar#1145]
Is there any chance to fix the code being blocked by the CSP in future releases?
https://web.dev/articles/csp?utm_source=devtools#unsafe-eval

We'd like to use Molstar but it goes against our SOC2 policies.

It would require removing instances of `eval(), new Function(), setTimeout([string], …), and setInterval([string], ...)`

🥺 🙏

**Comment by dsehnal** [comment]
Yes, it would be possible. There are only few cases of `new Function`:

- [setImmediate polyfill](https://github.com/molstar/molstar/blob/master/src/mol-task/util/scheduler.ts#L34) -- can be just removed as we don't use the string callback here
- [string interpolate function](https://github.com/molstar/molstar/blob/5af6a3e967937514a76783946046c272ffed6b2b/src/mol-util/string.ts#L69) 
  - [Binding.format](https://github.com/molstar/molstar/blob/5af6a3e967937514a76783946046c272ffed6b2b/src/mol-util/binding.ts#L51): this could be replaced by a simple `.replace('${triggers}', ...)`
  - [Swagger UI](https://github.com/molstar/molstar/blob/master/src/servers/common/swagger-ui/index.ts#L28): this could be modified to use basic string "replace all" too.
- [model](https://github.com/molstar/molstar/blob/master/src/servers/model/config.ts#L254) and [volume](https://github.com/molstar/molstar/blob/master/src/servers/volume/server/web-api.ts#L54) server id -> path mapping. Could be implemented without both `Function` and `interpolate`

Perhaps you can make a PR with it?

**Comment by arose** [comment]
@cmonnom care to do a PR?

**Comment by arose** [comment]
> @cmonnom care to do a PR?

still happy to accept a PR

**Comment by TaylorHo** [comment]
I've opened [this PR](https://github.com/molstar/molstar/pull/1922) to address the `new Function` usage. Can someone please review?
